Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  CardNumberViewModel.swift
+//  Frames
+//
+//  Created by Harry Brown on 07/07/2022.
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import UIKit
+import Checkout
+
+protocol CardNumberViewModelDelegate: AnyObject {
+  func update(cardNumber: String, scheme: Card.Scheme)
+}
+
+protocol CardNumberViewModelProtocol: TextFieldViewDelegate { }
+
+class CardNumberViewModel: CardNumberViewModelProtocol {
+  weak var delegate: CardNumberViewModelDelegate?
+  weak var cardNumberView: CardNumberViewProtocol?
+
+  private let cardUtils = CardUtils()
+  private let cardValidator: CardValidating
+
+  init(cardValidator: CardValidating) {
+    self.cardValidator = cardValidator
+  }
+
+  func buildView() -> CardNumberView {
```

Comment:
Not sure I have seen this flow previously:
- create viewModel
- ask view model to create view
- viewmodel registers itself on view, whilst also returning the newly created view.

Generally the expectation from most flows is:
```swift
let view = MyView()
let viewModel = MyViewModel()
viewModel.delegate = view // Should be a delegate for MyViewModelDelegate, enabling viewModel to ask UI to sync when needed
view.viewModel = viewModel
```

I am not too bothered about it for now, but I do worry a viewmodel normally should not have any UI knowledge. And in this case, the viewmodel file is importing UIKit, which is a red flag and can further weaken that separation over time.

Reply:
The inclusion of UIKit is only to implement the TextFieldViewDelegate methods. I'll sort out the setting of delegates

Reply:
Specifically for handling delegate methods, I wonder if we could pull that into a separate file (CardNumberViewModel+Extension.swift), where we only create an extension that conforms to the protocol. And then we can maybe have a cleaner separation 🤔

Reply:
This has been addressed now. CardNumberView handles anything UIKit specific and the business logic is still in the viewModel